src/actions/authActions.js

```diff
@@ -15,3 +17,9 @@ export const signupAction = (userData) => (dispatch) => backendCall.post('/users
   .catch((err) => {
     dispatch(authType(SIGNUP_FAIL, err.response.data));
   });
+
+export const saveUserToState = (userData) => (dispatch) => {
+  window.localStorage.setItem('token', userData.token);
+  dispatch(authType(SOCIAL_AUTH_PENDING, true));
+  dispatch(authType(SOCIAL_AUTH_SUCCESS, userData));
+};
```

Comment:
Give this function a more meaningful name. Also remove the pending type and move it to the component by creating an `isLoading` state